runtime/jcl/common/mgmtinit.c

```diff
@@ -886,22 +886,26 @@ initMemoryManagement(J9JavaVM *vm)
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_CLASSES;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_CLASSES);
 			segList = vm->classMemorySegments;
+			mgmt->nonHeapMemoryPools[idx].maxSize = -1;
 			break;
 		case MISC_MEMORY:
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_MISC;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_MISC);
 			segList = vm->memorySegments;
+			mgmt->nonHeapMemoryPools[idx].maxSize = -1;
 			break;
 		case JIT_CODECACHE:
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_JIT_CODE;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_JITCODE);
 			segList = vm->jitConfig->codeCacheList;
+			mgmt->nonHeapMemoryPools[idx].maxSize = vm->jitConfig->codeCacheTotalKB;
 			break;
 		case JIT_DATACACHE:
 		default:
 			mgmt->nonHeapMemoryPools[idx].id = J9VM_MANAGEMENT_POOL_NONHEAP_SEG_JIT_DATA;
 			strcpy(mgmt->nonHeapMemoryPools[idx].name, J9VM_MANAGEMENT_NONHEAPPOOL_NAME_JITDATA);
 			segList = vm->jitConfig->dataCacheList;
+			mgmt->nonHeapMemoryPools[idx].maxSize = vm->jitConfig->dataCacheTotalKB;
```

Comment:
This is unrelated to the current change, but I noticed that the `JIT_DATACACHE` & `default` are the same case.  Is this intended?

Reply:
currently JIT_DATACACHE is the last one for nonHeapMemoryPools, the default should not be a case here, maybe we should assert in the default case.

Reply:
+1 to asserting in the default case